### DISCUSSION TITLE
マルチプレイSDKでの利用条件表示とSDKの新しいバージョンのリリースについて

### DISCUSSION THREAD

**Discussion by nira-yusan** [pixiv/vroid-sdk-developers#133] [Q&A]
質問が二点あります。

・マルチプレイSDKを使った対戦プレイ時で相手側のモデルを読み込む際にも利用条件の表示は必要になるのでしょうか・・・？

・SDKは度々新しいバージョンがリリースされますが、その度に必ず新しいバージョンへの対応をする必要はありますか？

**Comment by nyamadan** [comment]
> ・マルチプレイSDKを使った対戦プレイ時で相手側のモデルを読み込む際にも利用条件の表示は必要になるのでしょうか・・・？

当のモデルを操作するユーザーに対して、利用条件の確認を必ず行っていただくようお願いいたします。厳密には、VRMパブリックライセンス文章の「モデル利用」「アバター利用」にあたる場合に、利用条件の確認をお願いしています。
対戦ゲームの場合、他のユーザー（対戦相手）が操作しているモデルの利用条件を表示する必要がありません

> ・SDKは度々新しいバージョンがリリースされますが、その度に必ず新しいバージョンへの対応をする必要はありますか？

バージョンを揃えるために差分がなくてもマルチプレイSDKをリリースすることがあります。
[CompatibilityVersion](https://developer.vroid.com/sdk/docs/0.4.0/Pixiv.VroidSdk.Version.html#Pixiv_VroidSdk_Version_CompatibilityVersion) をバージョンアップの目安として参照可能で、例えばVersionが0.2.0でCompatibilityVersionが0.1.0であれば0.1.0以降のマルチプレイSDKを使用することを想定しており、特にマルチプレイSDKをバージョンアップする必要はありません。

**Reply by nira-yusan** [reply]
対戦プレイに関しては、自分が使うモデルの条件を表示し、相手側の表示は必須ではないんですね。ありがとうございます。

通常のVRoidSDKも特にバージョンアップする必要はないのでしょうか？

**Reply by nyamadan** [reply]
> 通常のVRoidSDKも特にバージョンアップする必要はないのでしょうか？

バージョンアップにより機能追加のほかに不具合の修正なども行っています。
どのような不具合修正をしたか、どのような機能追加を行ったかは[GitHub](https://github.com/pixiv/vroid-sdk-developers/discussions/categories/announcements?discussions_q=is%3Aopen+category%3AAnnouncements)で確認できますのでこちらを参考にバージョンアップをご判断いただければと思います。

**Reply by nira-yusan** [reply]
つまり不具合修正などは行われるが、通常のSDKのバージョンアップへの対応は必須ではなく自己判断でOK、ということですね。ありがとうございます！

**Reply by nira-yusan** [reply]
すみません、

バージョンを揃えるために差分がなくてもマルチプレイSDKをリリースすることがあります。
CompatibilityVersionをバージョンアップの目安として参照可能で、例えばVersionが0.2.0でCompatibilityVersionが0.1.0であれば0.1.0以降のマルチプレイSDKを使用することを想定しており、特にマルチプレイSDKをバージョンアップする必要はありません。

と言われてましたが、これはVersionが必ずCompatibilityVersion以降のものである必要性があるのでしょうか？

**Reply by nira-yusan** [reply]
追記
CompatibilityVersionは各バージョンのVRoidSDKによって変わるものなのでしょうか？

CompatibilityVersionが0.2.0のVRoidSDKならバージョン0.2.0以降のマルチプレイSDKである必要がある。

・・・・・といった認識であっているでしょうか？

**Reply by nyamadan** [reply]
> CompatibilityVersionは各バージョンのVRoidSDKによって変わるものなのでしょうか？
> CompatibilityVersionが0.2.0のVRoidSDKならバージョン0.2.0以降のマルチプレイSDKである必要がある。

はい、その認識であっています。

> これはVersionが必ずCompatibilityVersion以降のものである必要性があるのでしょうか？

はい、 CompatibilityVersion をバージョンアップした場合、過去のMultiplaySDKとの互換性が失われていることを示します。
たとえば、CompatibilityVersionが0.2.0であるSDKに対してMultiplay SDK 0.1.0は使用できません。
